x-pack/plugins/apm/common/agent_configuration/setting_definitions/general_settings.ts

```diff
@@ -257,4 +442,76 @@ export const generalSettings: RawSettingDefinition[] = [
     ),
     includeAgents: ['java', 'nodejs', 'python', 'dotnet', 'ruby', 'go'],
   },
+
+  {
+    key: 'transaction_ignore_user_agents',
+    type: 'text',
+    defaultValue: '',
+    label: i18n.translate(
+      'xpack.apm.agentConfig.transactionIgnoreUserAgents.label',
+      {
+        defaultMessage: 'Transaction ignore user agents',
+      }
+    ),
+    description: i18n.translate(
+      'xpack.apm.agentConfig.transactionIgnoreUserAgents.description',
+      {
+        defaultMessage:
+          'Used to restrict requests from certain User-Agents from being instrumented.\n' +
+          '\n' +
+          'When an incoming HTTP request is detected,\n' +
+          'the User-Agent from the request headers will be tested against each element in this list.\n' +
+          'Example: `curl/*`, `*pingdom*`',
+      }
+    ),
+    includeAgents: ['java'],
+  },
+
+  {
+    key: 'use_path_as_transaction_name',
+    type: 'boolean',
+    defaultValue: 'false',
+    label: i18n.translate(
+      'xpack.apm.agentConfig.usePathAsTransactionName.label',
+      {
+        defaultMessage: 'Use path as transaction name',
+      }
+    ),
+    description: i18n.translate(
+      'xpack.apm.agentConfig.usePathAsTransactionName.description',
+      {
+        defaultMessage:
+          'If set to `true`,\n' +
+          'transaction names of unsupported or partially-supported frameworks will be in the form of `$method $path` instead of just `$method unknown route`.\n' +
+          '\n' +
+          'WARNING: If your URLs contain path parameters like `/user/$userId`,\n' +
+          'you should be very careful when enabling this flag,\n' +
+          'as it can lead to an explosion of transaction groups.\n' +
+          'Take a look at the `transaction_name_groups` option on how to mitigate this problem by grouping URLs together.',
+      }
+    ),
+    includeAgents: ['java'],
+  },
+
+  {
+    key: 'transaction_name_groups',
+    type: 'text',
+    defaultValue: '',
+    label: i18n.translate('xpack.apm.agentConfig.transactionNameGroups.label', {
+      defaultMessage: 'Transaction name groups',
+    }),
+    description: i18n.translate(
+      'xpack.apm.agentConfig.transactionNameGroups.description',
+      {
+        defaultMessage:
+          'With this option,\n' +
+          'you can group transaction names that contain dynamic parts with a wildcard expression.\n' +
+          'For example,\n' +
+          'the pattern `GET /user/*/cart` would consolidate transactions,\n' +
+          'such as `GET /users/42/cart` and `GET /users/73/cart` into a single transaction name `GET /users/*/cart`,\n' +
+          'hence reducing the transaction name cardinality.',
+      }
+    ),
+    includeAgents: ['java'],
```

Comment:
If these settings are java specific they should go in [java_settings.ts](https://github.com/elastic/kibana/blob/9d819bac922a48d77f627ab6cba67e9b54ad96b8/x-pack/plugins/apm/common/agent_configuration/setting_definitions/java_settings.ts). If general, why are they only made available for the java agent?

Reply:
Not all of these options are exclusive for the Java agent. I know some of these are implemented in other agents, or at least could potentially be implemented by other agents. The idea is that as a next step, the agent teams will update this to include their agent if it supports the option.